remo/remozilla/models.py

```diff
@@ -127,26 +123,36 @@ def save(self, *args, **kwargs):
         # Update action items
         action_model = ContentType.objects.get_for_model(self)
         if self.pk:
+            # Get saved action item
+            action_items = ActionItem.objects.filter(
+                content_type=action_model, object_id=self.pk, resolved=False)
+            # If there is no user or user is not rep or the bug is resolved,
+            # resolve the action item too!
             if (not self.assigned_to or not user_is_rep(self.assigned_to) or
                     self.status == 'RESOLVED'):
-                items = ActionItem.objects.filter(content_type=action_model,
-                                                  object_id=self.pk)
-                items.update(resolved=True)
+                action_items.update(resolved=True)
             else:
-                actions = [ADD_RECEIPTS_ACTION, ADD_REPORT_ACTION,
-                           ADD_PHOTOS_ACTION, ADD_REPORTS_PHOTOS_ACTION,
-                           REVIEW_BUDGET_REQUEST_ACTION,
-                           WAITING_MENTOR_VALIDATION_ACTION]
+                possible_actions = [ADD_RECEIPTS_ACTION, ADD_REPORT_ACTION,
+                                    ADD_PHOTOS_ACTION,
+                                    ADD_REPORTS_PHOTOS_ACTION,
+                                    REVIEW_BUDGET_REQUEST_ACTION,
+                                    WAITING_MENTOR_VALIDATION_ACTION]
                 action_names = (['{0} {1}'.format(action, self.summary)
-                                 for action in actions])
-
+                                 for action in possible_actions])
+                bug_attrs = ['waiting_receipts', 'waiting_report',
```

Comment:
I suggest you move this outside this class and use it as a constant. It's probably better to be under "remozilla" app.